lib/spree/adyen/invoice.rb

```diff
@@ -63,7 +63,7 @@ def pre_tax_amount_from_line_item line_item
         amount = (line_item.discounted_amount - line_item.included_tax_total) / line_item.quantity
 
         Spree::Money.new(
-          BigDecimal.new(amount.to_s).round(2, BigDecimal::ROUND_HALF_DOWN),
+          BigDecimal(amount.to_s).round(2, BigDecimal::ROUND_HALF_DOWN),
```

Comment:
The updates in this file are the only actual updates for Ruby 2.7. The rest is piecing together compatible gems and code changes to get the specs working again.